src/Styles.c

```diff
@@ -1212,7 +1212,7 @@ void Style_SetLexer(HWND hwnd, PEDITLEXER pLexNew)
 
   if (StrStr(pCurrentStandard->Styles[STY_CARET].szValue, L"block")) {
     StringCchCat(wchSpecificStyle, COUNTOF(wchSpecificStyle), L"; block");
-    SendMessage(hwnd, SCI_SETCARETSTYLE, (CARETSTYLE_BLOCK | ovrstrk_mode), 0);
+    SendMessage(hwnd, SCI_SETCARETSTYLE, (CARETSTYLE_BLOCK_AFTER | ovrstrk_mode), 0);

```

Comment:
This is wrong, `CARETSTYLE_BLOCK_AFTER` don't set bock caret style (`CARETSTYLE_BLOCK`), it's flag to draw block caret out selection.

Reply:
This is not wrong, cause it is a valid bit for the caret style, but it was not intended here ;-)

Reply:
See the code on the only usage of CARETSTYLE_BLOCK_AFTER in ViewStyle::DrawCaretInsideSelection().

It was designed to a flag to draw block caret outside selection. The block caret (shape) still need to be enabled by CARETSTYLE_BLOCK. The next method ViewStyle::CaretShapeForMode() defined the caret shape, which don't use CARETSTYLE_BLOCK_AFTER  at all.